src/components/layout/GradientBottomActions.tsx

```diff
@@ -1,18 +1,20 @@
 import * as React from "react";
-import { View, StyleSheet, ViewStyle, StyleProp } from "react-native";
+import { StyleProp, StyleSheet, View, ViewStyle } from "react-native";
 import LinearGradient from "react-native-linear-gradient";
 import Animated from "react-native-reanimated";
+import { IOColors, IOSpacer, IOVisualCostants, hexToRgba } from "../../core";
 import { WithTestID } from "../../utils/types";
-import { IOColors, hexToRgba, IOSpacer, IOVisualCostants } from "../../core";
+import { ButtonLink, ButtonSolid } from "../buttons";
 import { VSpacer } from "../spacer";
+import { GradientBottomAction } from "./common";
 
 export type GradientBottomActions = WithTestID<{
   transitionAnimStyle: Animated.AnimateStyle<StyleProp<ViewStyle>>;
   dimensions: GradientBottomActionsDimensions;
   // Accepted components: ButtonSolid, ButtonLink
   // Don't use any components other than this, please.
-  primaryAction?: React.ReactNode;
-  secondaryAction?: React.ReactNode;
+  primaryAction?: GradientBottomAction;
+  secondaryAction?: GradientBottomAction;
```

Comment:
We can just set the relative types, without being generic:
```suggestion
  primaryAction?: ButtonSolidAction;
  secondaryAction?: ButtonLinkAction;
```